lib/google.js

```diff
@@ -137,15 +159,58 @@ module.exports.getPurchaseData = function (purchase) {
 		return null;
 	}
 	var data = [];
-	data.push({
-		transactionId: purchase.orderId,
+	var purchaseInfo = {
 		productId: purchase.productId,
 		purchaseDate: purchase.purchaseTime,
 		quantity: 1
-	});
+	};
+	
+	if(checkSubscriptionState){
+		purchaseInfo.transactionId = purchase.purchaseToken;
+		purchaseInfo.expirationDate = purchase.expirationTime;
+	}
+
+	data.push(purchaseInfo);
 	return data;
 };
 
+/**
+ * Function to check subscription status in Google Play
+ * @param  {Object}   	data 		receipt data
+ * @param  {Function} 	cb   		callback function
+ */
+function checkSubscriptionStatus(data, cb){
+	
+	data.service = constants.SERVICES.GOOGLE;
+
+	if(checkSubscriptionState){
+
+		var packageName = data.packageName;
+		var subscriptionID = data.productId;
+		var purchaseToken = data.purchaseToken;
+
+		var url = "https://www.googleapis.com/androidpublisher/v2/applications/"+packageName+"/purchases/subscriptions/"+subscriptionID+"/tokens/"+purchaseToken;
+
+		getSubscriptionInfo(url, function(error, response, body){
```

Comment:
I use jshint to lint my code. There is a shell script in the root of the project, "lint.sh".
If you execute `make`, it creates git pre-commit hook with linting. Please make sure you do that before committing the code.